libraries/chain/block_log.cpp

```diff
@@ -603,6 +619,43 @@ namespace eosio { namespace chain {
             return pos;
          }
 
+         block_pos_size_t get_block_position_and_size(uint32_t block_num) {
```

Comment:
For this and the other added methods can you add unittests for them.

Reply:
Can the tests be in a separate PR. I plan to write some comprehensive tests.

Reply:
There is no rush to get this in, but doesn't matter to me.

Reply:
https://github.com/AntelopeIO/spring/issues/808